Xamarin.Forms.Core/FontSizeConverter.cs

```diff
@@ -4,44 +4,63 @@
 
 namespace Xamarin.Forms
 {
-	[Xaml.TypeConversion(typeof(double))]
+	[TypeConversion(typeof(double))]
 	public class FontSizeConverter : TypeConverter, IExtendedTypeConverter
 	{
 		[Obsolete("IExtendedTypeConverter.ConvertFrom is obsolete as of version 2.2.0. Please use ConvertFromInvariantString (string, IServiceProvider) instead.")]
 		object IExtendedTypeConverter.ConvertFrom(CultureInfo culture, object value, IServiceProvider serviceProvider)
-		{
-			return ((IExtendedTypeConverter)this).ConvertFromInvariantString(value as string, serviceProvider);
-		}
+			=> ((IExtendedTypeConverter)this).ConvertFromInvariantString(value as string, serviceProvider);
 
 		object IExtendedTypeConverter.ConvertFromInvariantString(string value, IServiceProvider serviceProvider)
 		{
-			if (value != null)
-			{
-				double size;
-				if (double.TryParse(value, NumberStyles.Number, CultureInfo.InvariantCulture, out size))
+			if (value != null) {
+				value = value.Trim();
+				if (double.TryParse(value, NumberStyles.Number, CultureInfo.InvariantCulture, out double size))
 					return size;
+
 				var ignoreCase = (serviceProvider?.GetService(typeof(IConverterOptions)) as IConverterOptions)?.IgnoreCase ?? false;
-				NamedSize namedSize;
-				if (Enum.TryParse(value, ignoreCase, out namedSize))
-				{
-					Type type;
-					var valueTargetProvider = serviceProvider.GetService(typeof(IProvideValueTarget)) as IProvideValueTarget;
-					type = valueTargetProvider != null ? valueTargetProvider.TargetObject.GetType() : typeof(Label);
-					return Device.GetNamedSize(namedSize, type, false);
+				var sc = ignoreCase ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal;
+				NamedSize? namedSize = null;
+
+				if (value.Equals("Default", sc))
```

Comment:
Shouldn't we use `nameof(NamedSize.Default)`?

Reply:
fixed